tests/utils.test.ts

```diff
@@ -98,21 +98,28 @@ describe("Utils", () => {
 
   describe("toBigInt", () => {
     it("should return Big Int with 8 decimals by default", () => {
-      expect(toBigInt(12.5345)).toBe(1_253_450_000);
+      expect(toBigInt(12.5345)).toStrictEqual(1_253_450_000);
+      expect(toBigInt(12.5345) % 1).toStrictEqual(0);
     });
 
     it("should return Big Int with decimals passed in param", () => {
-      expect(toBigInt(12.5345, 6)).toBe(12_534_500);
+      expect(toBigInt(12.5345, 6)).toStrictEqual(12_534_500);
+      expect(toBigInt(12.5345, 6) % 1).toStrictEqual(0);
+    });
+
+    it("should return rounded Big Int with decimals passed in param", () => {
+      expect(toBigInt(120139.69456927, 7)).toStrictEqual(1_201_396_945_693);
```

Comment:
```
expect(toBigInt(120139.69456927, 7)).toStrictEqual(1_201_396_945_692);
expect(toBigInt(94.03999999999999, 8)).toStrictEqual(9_403_999_999);
```

Reply:
Ok, I didn't see that you shouldn't round up.
I tried another approach by converting to a character string.
It passes the tests but maybe there are other tests to add?